packages/core/react/src/WalletProviderBase.tsx

```diff
@@ -8,54 +8,58 @@ import type {
 } from '@solana/wallet-adapter-base';
 import { WalletNotConnectedError, WalletNotReadyError, WalletReadyState } from '@solana/wallet-adapter-base';
 import type { PublicKey } from '@solana/web3.js';
-import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import React, { type ReactNode, useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { WalletNotSelectedError } from './errors.js';
 import { WalletContext } from './useWallet.js';
-import type { WalletProviderProps } from './WalletProvider.js';
 
-type Props = Readonly<
-    Omit<WalletProviderProps, 'autoConnect' | 'localStorageKey'> & {
-        adapter: Adapter | null;
-        isUnloadingRef: React.RefObject<boolean>;
-        // NOTE: The presence/absence of this handler implies that auto-connect is enabled/disabled.
-        onAutoConnectRequest?: () => Promise<void>;
-        onConnectError: () => void;
-        onSelectWallet: (walletName: WalletName | null) => void;
-    }
->;
+export interface WalletProviderBaseProps {
+    children: ReactNode;
+    wallets: Adapter[];
+    adapter: Adapter | null;
+    isUnloadingRef: React.RefObject<boolean>;
+    // NOTE: The presence/absence of this handler implies that auto-connect is enabled/disabled.
+    onAutoConnectRequest?: () => Promise<void>;
+    onConnectError: () => void;
+    onError?: (error: WalletError, adapter?: Adapter) => void;
+    onSelectWallet: (walletName: WalletName | null) => void;
+}
 
 export function WalletProviderBase({
-    adapter,
     children,
+    wallets: adapters,
+    adapter,
     isUnloadingRef,
     onAutoConnectRequest,
     onConnectError,
-    onError,
+    onError = (error, adapter) => {
```

Comment:
I _think_ this will result in a new copy of this function on every render, which will thrash the effect.

Reply:
Ahh, you're probably right.